jerry-core/ecma/base/ecma-globals.h

```diff
@@ -115,6 +120,26 @@ typedef uint32_t ecma_value_t;
 #define ECMA_VALUE_SHIFT 3
 
 /**
+ * Flag for simple values
+ */
+#define ECMA_SIMPLE_VALUE_FLAG 0x8u
+
+/**
+ * Shift for integer value part in ecma_value_t
+ */
+#define ECMA_INTEGER_NUMBER_SHIFT 4
```

Comment:
Terminology(?): the use of this macro at `ecma_make_simple_value` reveals that this shift is not specific to integers but to all directly encoded (i.e., non-prt) values. Thus, I think that `ECMA_DIRECT_SHIFT` would be a better name.